From ea51a7c6577d641a6771e6f1ae52bc78628c340f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20Dre=C3=9Fler?= <verdre@v0yd.nl>
Date: Thu, 20 Jun 2019 21:01:52 +0000
Subject: [PATCH] altTab: Ensure style of this._list before calculating icon
 sizes

We're calculating icon sizes for the alt tab switcher early and at a
point where the style attributes of this._list are not loaded yet. To
make sure the value of this._list.spacing is correct, call
ensure_style() on this._list before accessing the spacing.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/587


(cherry picked from commit eab320dab59e702cda3d2864be2bb59c03d34d79)
---
 js/ui/altTab.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/js/ui/altTab.js b/js/ui/altTab.js
index fb9c875c8..ad2c0ca7a 100644
--- a/js/ui/altTab.js
+++ b/js/ui/altTab.js
@@ -730,6 +730,7 @@ class AppSwitcher extends SwitcherPopup.SwitcherList {
                 j++;
         }
         let themeNode = this._items[j].get_theme_node();
+        this._list.ensure_style();
 
         let iconPadding = themeNode.get_horizontal_padding();
         let iconBorder = themeNode.get_border_width(St.Side.LEFT) + themeNode.get_border_width(St.Side.RIGHT);
-- 
2.20.1

